### PR TITLE
Upgrade to `mosaicml-streaming==0.5.x`

### PR DESCRIPTION
**Pull request by abhi-mosaic**
- updates streaming version
- enables arg checking via passing in args as **kwargs

### CODE REVIEW

File: llmfoundry/data/text_data.py
Comment (vchiley) [inline llmfoundry/data/text_data.py:238]:
```suggestion
                    choose=stream.get('choose', None),
```

**Review by vchiley — APPROVED** [review]
Would it make sense to run 125M with v0.4 and v0.5 to verify training looks the same?

Approving to unblock. Generally lgtm

**Comment by abhi-mosaic** [comment]
Yup will run some convergence tests before merging

File: llmfoundry/data/text_data.py
Comment (dakinggg) [inline llmfoundry/data/text_data.py:241]: given the way we are accessing these args with `.get`, can we add some simple validation based on streaming version and the `stream` cfg dict? i know we have a full validation jira planned, but i think this one is likely to bite people.

Comment (abhi-mosaic) [reply llmfoundry/data/text_data.py:241]: Yeah I can add some kind of validation, these `.get`s worry me too

**Comment by abhi-mosaic** [comment]
Would recommend waiting on this merge until we get a hotfix release 0.5.1

File: llmfoundry/data/text_data.py
Comment (vchiley) [reply llmfoundry/data/text_data.py:241]: Would [**kwargs handling kwarg checking](https://github.com/mosaicml/llm-foundry/pull/292/commits/bd57a87a5ade3bc4e83fac02316ecf506040e540) do the trick?

**Comment by vchiley** [comment]
With baseline usage, main and this branch have similar performance
Screenshot 2023-06-20 at 4 06 34 PM
MFU is also effectively exactly the same

the only diff is that the update uses slightly less sys mem
Screenshot 2023-06-20 at 4 08 28 PM

**Pull request merged**
